src/main/java/com/minecolonies/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -732,10 +740,13 @@ private void clearActionsDone()
      * Dumps one inventory slot into the building chest.
      *
      * @param keepIt used to test it that stack should be kept
+     * @param keptX items already kept
      * @return true if is has to dump more.
      */
-    private boolean dumpOneMoreSlot(@NotNull Predicate<ItemStack> keepIt)
+    private boolean dumpOneMoreSlot(@NotNull Predicate<ItemStack> keepIt, Map<ItemStorage, Integer> keptX)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "dumpOneMoreSlot" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)